### ISSUE TITLE
height calculation of stacked items is off if no item is visible in a line currently

### ISSUE THREAD

**Issue by Felix-N** [namespace-ee/react-calendar-timeline#497]
the calculated lineHeight is off when stacking is enabled and a given line has no item visible (or on the canvas).

If an item comes into the view / canvas the line hight gets calculated. However the linehight is incorrect.

I've fixed it for my use case and added it to the pull request 
https://github.com/namespace-ee/react-calendar-timeline/pull/495

Though I've just updated the snapshots and did not check if all cases are handled correctly.

**Comment by Ilaiwi** [comment]
@Felix-N could you please provide a gif for this?

**Comment by Felix-N** [comment]
Not a gif but 2 images:
lineHeight is set to 25
<img width="1176" alt="1" src="https://user-images.githubusercontent.com/1209409/51478078-b8861c80-1d8a-11e9-8845-9fdb857f0a4c.png">
<img width="1303" alt="2" src="https://user-images.githubusercontent.com/1209409/51478079-b8861c80-1d8a-11e9-9c73-f29eaeb4d39e.png">

The changes to groupStack to the verticalMargin fixed the issue for me (they are in the pull request as well).

Here my current version as copy/paste
```
/**
 * Calculate the position of a given item for a group that
 * is being stacked
 */
export function groupStack(
  lineHeight,
  item,
  group,
  groupHeight,
  groupTop,
  itemIndex
) {
  // calculate non-overlapping positions
  let curHeight = groupHeight
  let verticalMargin = (lineHeight - item.dimensions.height) / 2
  if (item.dimensions.stack && item.dimensions.top === null) {
    item.dimensions.top = groupTop + verticalMargin
    curHeight = Math.max(curHeight, lineHeight)
    do {
      var collidingItem = null
      //Items are placed from i=0 onwards, only check items with index < i
      for (var j = itemIndex - 1, jj = 0; j >= jj; j--) {
        var other = group[j]
        if (
          other.dimensions.top !== null &&
          other.dimensions.stack &&
          collision(item.dimensions, other.dimensions, lineHeight)
        ) {
          collidingItem = other
          break
        } else {
          // console.log('dont test', other.top !== null, other !== item, other.stack);
        }
      }

      if (collidingItem != null) {
        // There is a collision. Reposition the items above the colliding element
        item.dimensions.top = collidingItem.dimensions.top + lineHeight
        curHeight = Math.max(
          curHeight,
          item.dimensions.top + item.dimensions.height + verticalMargin - groupTop
        )
      }
    } while (collidingItem)
  }
  return {
    groupHeight: curHeight,
    verticalMargin,
    itemTop: item.dimensions.top
  }

}
```

**Comment by Ilaiwi** [comment]
@Felix-N  sorry for the late response I investigated your issue and you are correct. The heights are inconsistent and your fix resolves the issues. Thank you.

Although could you please make a pull request with only the fix for this issue and remove the other changes?

I want to give you the credit of your fix but I can't merge it with the other changes.

**Comment by Ilaiwi** [comment]
released in 0.23.1
